compiler/crates/relay-codegen/src/build_ast.rs

```diff
@@ -142,10 +222,12 @@ impl<'schema, 'builder> CodegenBuilder<'schema, 'builder> {
     }
 
     fn build_operation(&mut self, operation: &OperationDefinition) -> AstKey {
+        let definition = ExecutableDefinition::Operation(operation.clone());
```

Comment:
cloning is no good. what are some ways to avoid this clone?

- create the artifact location here and pass that down instead? feels a little bit more targeted but it's nice to have the definition available to you. 
- use an Arc instead?

Reply:
How about we take just the source location key and set it as a property on the `CodegenBuilder` struct?

So `CodegenBuilder` would have a new field `definition_source_location: Option<WithLocation<StringKey>>` which `build_operation` and `build_fragment` would be responsible for setting/unsetting before/after traversing their selections.

This would avoid the noisy threading of `definition` through all the methods.

Reply:
Nice, feels much better. I `definition_source_location` doesn't even have to be optional since we construct a new builder for each definition.